packages/chain/statemgr/syncmgr.go

```diff
@@ -118,6 +127,35 @@ func (sm *stateManager) doSyncActionIfNeeded() {
 	}
 }
 
+func (sm *stateManager) candidateBlockInWAL(i uint32) (blockAdded, blockApproved bool) {
```

Comment:
No need to check if new block is approved
```suggestion
func (sm *stateManager) candidateBlockInWAL(i uint32) (blockAdded bool) {
```